tensorboard/webapp/app_routing/effects/app_routing_effects_test.ts

```diff
@@ -255,6 +277,94 @@ describe('app_routing_effects', () => {
       ]);
     });
 
+    describe('dispatchNavigating$ with dirty updates', () => {
+      beforeEach(() => {
+        const activeRoute = buildRoute({
+          routeKind: RouteKind.EXPERIMENTS,
+          pathname: '/experiments',
+          queryParams: [],
+          navigationOptions: {
+            replaceState: false,
+          },
+        });
+        const dirtyExperimentsFactory = () => {
+          return {experimentIds: ['otter']};
+        };
+
+        store.overrideSelector(getActiveRoute, activeRoute);
+        store.overrideSelector(
+          testDirtyExperimentsSelector,
+          dirtyExperimentsFactory()
+        );
+        store.refreshState();
+      });
+
+      it('warns user when navigating away', () => {
+        spyOn(window, 'confirm');
+        action.next(
+          actions.navigationRequested({
+            pathname: '/experiment/meow',
+          })
+        );
+
+        expect(window.confirm).toHaveBeenCalledWith(
+          'You have unsaved edits, are you sure you want to discard them?'
+        );
+      });
+
+      it('noops if user cancels navigation', () => {
+        spyOn(window, 'confirm').and.returnValue(false);
+        action.next(
+          actions.navigationRequested({
+            pathname: '/experiment/meow',
+          })
+        );
+
+        expect(window.confirm).toHaveBeenCalledTimes(1);
+        expect(actualActions).toEqual([]);
+      });
+
+      it('fires discardDirtyUpdates, navigating and navigated if user discards dirty updates', () => {
+        fakeAsync(() => {
+          spyOn(window, 'confirm').and.returnValue(true);
+          action.next(
+            actions.navigationRequested({
+              pathname: '/experiment/meow',
+            })
+          );
+
+          expect(window.confirm).toHaveBeenCalledTimes(1);
+          expect(actualActions).toEqual([actions.discardDirtyUpdates()]);
+
+          tick();
+          expect(actualActions).toEqual([
+            actions.navigating({
```

Comment:
I am surprised that the test is passing here. Shouldn't the `actualActions` be an array with two elements where the first element in the `actions.dsicardDirtyUpdates()`? Similarly, I  expected the L352 to have three elements.  What am I missing?

Reply:
I used `fakeAsync` incorrectly before and it didn't capture the errors. Fixed.